compiler-core/src/io.rs

```diff
@@ -164,6 +164,7 @@ pub trait FileSystemReader {
     fn read_bytes(&self, path: &Path) -> Result<Vec<u8>, Error>;
     fn reader(&self, path: &Path) -> Result<WrappedReader, Error>;
     fn is_file(&self, path: &Path) -> bool;
+    fn is_valid_module_path(&self, path: &Path, dir: &Path) -> bool;
```

Comment:
This doesn't perform IO so it shouldn't be on the IO trait that gets injected in, rather it should be a regular function in a module.